### ISSUE TITLE
Can't get Demo To Work

### ISSUE THREAD

**Issue by gitpushpray1** [RangerMauve/mutable-webtorrent#3]
```
Created Keypair: Public [  336...624f ] Secret [  b0c3...629 ]
Preparing to seed /mnt/c/code/mwt/stufftoshare
Created torrent 17e1...902f0 [ 'example/example.txt' ]
Publishing torrent
Published torrent magnet:?xs=urn:btpk:336f...624f - Version 0
Resolved latest version: { infoHash: <Buffer 17 e1 c1 ... 92 29 02 f0>,
  sequence: 0 }
Republishing version to the DHT
Republishing version to the DHT
Republishing version to the DHT
...
```

Then ran the download.js file:
```
node download.js magnet:?xs=urn:btpk:336f...624f
```

I got an error:
```
/mnt/c/code/mwt/node_modules/mutable-webtorrent/index.js:47
    const parsed = new URL(magnetURI)
                   ^

ReferenceError: URL is not defined
    at MutableWebTorrent.add (/mnt/c/code/mwt/node_modules/mutable-webtorrent/index.js:47:20)
```

I cloned the index.js locally to add the missing package with a few console.log statements, and the application hangs.

I tried taking the magnet link and inserting it into a BT client to download, and it's throwing a parse exception. 

Am I missing something to get the demo working?

**Comment by RangerMauve** [comment]
What version of Node are you using? Does the BT client you're using support mutable torrents?

**Comment by gitpushpray1** [comment]
```
node --version
v8.16.0
```

Using uTorrent 3.5.5  Looking to see if it supports mutable torrents. 

Do you know a better bt client to test with?

**Comment by RangerMauve** [comment]
I'm not sure if any client supports mutable torrents, actually.

Try upgrading Node to at least Node 10. Or better yet, node 11.

**Comment by gitpushpray1** [comment]
Updated using the following:

```
sudo npm cache clean -f
sudo npm install -g n
sudo n stable
```

Restart terminal:
```
node --version
v10.16.2
```

Still hanging. I did get a null exception on download at: https://github.com/RangerMauve/mutable-webtorrent/blob/master/index.js#L83 

I restarted, and I went back to hanging. 

Resolve function response looks like:

```
{ id:
   <Buffer 3c f4 ... 99 81>,
  k:
   <Buffer 04 6b ... c7 08>,
  nodes:
   <Buffer 3c f4 ... c6 83 ... >,
  p: 52186,
  seq: 0,
  sig:
   <Buffer fe 7d ... 4a d1 ... >,
  token: <Buffer 58 79 17 4a>,
  v:
   { ih:
      <Buffer 17 e1 ... 02 f0> } }
```

**Comment by RangerMauve** [comment]
So, just to clarify, you have one process seeding a torrent and another process downloading it?

I'm not sure, but it might be that it doesn't work well on the same computer. I think when I was testing it I had the seeder on a Digital Ocean droplet and downloaded using my laptop.

**Comment by gitpushpray1** [comment]
>So, just to clarify, you have one process seeding a torrent and another process downloading it?

Yes.

Tonight I'll pull out my laptop and attempt seeding from Desktop and downloading on laptop.

**Comment by gitpushpray1** [comment]
Pulled out the laptop. Have desktop seeding the directory, and laptop has the download script. Download is still unsuccessful. 

Desktop shows:
```
Created torrent 22d8....b21b [ 'example/example.txt' ]
...
{ id:
   <Buffer be a2 ... 83 82>,
  k:
   <Buffer fa 61 ... 49 54>,
  nodes:
   <Buffer be a2 ... df 93 ... >,
  p: 51024,
  seq: 0,
  sig:
   <Buffer 5d 64 ... a8 fc ... >,
  token: <Buffer 2c 2c 90 c9>,
  v:
   { ih:
      <Buffer 22 d8 ... b2 1b> } }
Resolved latest version: { infoHash:
   <Buffer 22 d8 ... b2 1b>,
  sequence: 0 }
```

Created cl1p to send the data to my laptop.

Laptop shows: 
```
node --version
v12.6.0
```

```
{
  id: <Buffer be a2 ... 83 82>,
  k: <Buffer fa 61 ... 49 54>,
  nodes: <Buffer be a2 ... df 93 ... 158 more bytes>,
  p: 64439,
  seq: 0,
  sig: <Buffer 5d 64 ... a8 fc ... 14 more bytes>,
  token: <Buffer e4 8c 5b fe>,
  v: {
    ih: <Buffer 22 d8 ... b2 1b>
  }
}
```

Laptop has download directory at package level which is empty, and the application is hanging.

**Comment by RangerMauve** [comment]
Dang. I'm not sure what's up then. I've got a lot of stuff on my plate at the moment, sadly. But I might have time to look at this more towards the end of the month.

**Comment by jumpersdevice** [comment]
did you happen to test it more or got it working by any chance?

thanks

**Comment by RangerMauve** [comment]
Sadly I've been unable to devote any more time to this module. 😅 I might have time to take on support contracts over the summer if anyone is interested though.

**Comment by RangerMauve** [comment]
I've just published a new version of mutable-webtorrent@1.2.0 and replaced the ed25519-supercop module with sodium-universal which seems to be more actively maintained.

Running it with node 12.16.2 seems to work fine. Gonna close this for now unless people have more issues with a recent node version.
